akka-actor-tests/src/test/scala/akka/util/ByteStringSpec.scala

```diff
@@ -365,13 +365,30 @@ class ByteStringSpec extends WordSpec with Matchers with Checkers {
       ByteStrings(ByteString1.fromString(""), ByteString1.fromString("ab")).dropRight(Int.MinValue) should ===(ByteString("ab"))
     }
     "slice" in {
-      ByteStrings(ByteString1.fromString("a"), ByteString1.fromString("")).slice(0, 1) should ===(ByteString("a"))
       ByteStrings(ByteString1.fromString(""), ByteString1.fromString("a")).slice(1, 1) should ===(ByteString(""))
+      // Can obtain expected results from 6 basic patterns
+      ByteStrings(ByteString1.fromString("a"), ByteString1.fromString("")).slice(-10, 10) should ===(ByteString("a"))
```

Comment:
Yeah and note that these do not cover ByteString1.slice, so I suggested a few additional ones.
Technically ScalaCheck does cover them, but in practice these tests are much easier to debug/track.

To find more edge cases it's easy to use the below scalacheck config I pasted